### PR TITLE
Recover some items that expect braces and don't take semicolons

### LINKED ISSUE #102262: Recover from incorrect use of semicolon in various empty declaration/impl
Given the following code: [link](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2021&gist=3d59ad6ba33e68b8aa1997d993b2cea3)

```rust
trait Trait;

impl Trait for ();

enum Enum;

fn main() {

}

```

The current output is:

```
   Compiling playground v0.0.1 (/playground)
error: expected one of `:`, `<`, `=`, `where`, or `{`, found `;`
 --> src/main.rs:1:12
  |
1 | trait Trait;
  |            ^ expected one of `:`, `<`, `=`, `where`, or `{`

error: could not compile `playground` due to previous error
```

Ideally the output should suggest adding `{}` for each of the incorrect empty declaration/impl shown above. Thanks.

### PR DESCRIPTION
**Pull request by compiler-errors**
Fixes #102262

### CODE REVIEW

**Review by JohnTitor — APPROVED** [review]
Makes sense to me 👍

File: compiler/rustc_error_messages/locales/en-US/parser.ftl [hunk 1]
```diff
@@ -158,3 +158,6 @@ parser_remove_let = expected pattern, found `let`
 
 parser_use_eq_instead = unexpected `==`
     .suggestion = try using `=` instead
+
+parser_use_empty_block_not_semi = expected { "`{}`" }, found `;`
+    .suggestion = try using { "`{}`" } instead
```
Comment (Rageking8) [inline compiler/rustc_error_messages/locales/en-US/parser.ftl:163]: Not sure if "try using `{}` instead of `;`" is better or worse?

Comment (compiler-errors) [reply compiler/rustc_error_messages/locales/en-US/parser.ftl:163]: found it kinda redundant given the primary message of the error, but i don't care either way

Comment (Rageking8) [reply compiler/rustc_error_messages/locales/en-US/parser.ftl:163]: Ok sure.

**Review by davidtwco — APPROVED** [review]

**Pull request merged**